src/routes/(marketing)/(auth)/register/+page.server.ts

```diff
@@ -39,12 +39,23 @@ export const actions: Actions = {
 
 		if (error) {
 			console.error(error);
+
+			// Check to see if sign-ups are disabled in Supabase
+			if (error.code === 'signup_disabled' || error.message?.includes('Signups not allowed')) {
+				return {
+					form,
+					signupDisabled: true
+				};
+			}
+
 			return setError(form, '', 'Could not sign up. Please try again.');
 		}
 
-		const search = new URLSearchParams(event.url.search);
-		search.set('next', event.url.searchParams.get('next') || '/dashboard');
-
-		redirect(303, '/auth/callback?' + search.toString());
+		// Instead of redirecting, return success status and the email used to sign up
```

Comment:
We can check if `session` returned from `signUp` is non-null. If so, we can still redirect.